WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/visibility/PrepublishingVisibilityFragment.kt

```diff
@@ -47,7 +47,7 @@ class PrepublishingVisibilityFragment : Fragment(), PostSettingsInputDialogListe
 
     override fun onResume() {
         super.onResume()
-        viewModel.setEditPostRepository(getEditPostRepository())
+        viewModel.setEditPostRepositoryAfterConfigChanges(getEditPostRepository())
```

Comment:
@ashiagr let me know what you think about this approach. The reason I had to do this is, when you are entering the password in the dialog and rotation occurs the `ViewModel` retains the `editPostRepository` but it's instance is now different from the one in `EditPostActivity` so when the password is on the retained instance and you press back the state is lost so here I am setting it so that it's consistent. Let me know if this approach is okay or if anything can be simplified. For more clarity, I am passing the `editPostRepository` in `start` as well. Thanks 😄

Reply:
As start method is called every time our fragment is recreated (which happens even when orientation changes), how about we move `editPostRepository` assignment above `isStarted` check to avoid calling `setEditPostRepositoryAfterConfigChanges` from fragment's `onResume`? 

```
fun start(editPostRepository: EditPostRepository) {
    this.editPostRepository = editPostRepository // <- here
    if (isStarted) return
    isStarted = true

    setToolbarUiState()
    updateUiState()
}
```

Reply:
Yes, thanks! This is a good approach. I was wondering if it would be okay to put code above the `isStarted` if that would break it's purpose so thanks for clearing that up.